docs: unify wallet transfer docs (#2573)
FuelLabs/fuels-ts: apps/docs/src/guide/wallets/wallet-transferring.md

## Transferring Assets To Contracts

When transferring assets to a deployed contract, we use the `transferToContract` method, which shares a similar parameter structure with the `transfer` method.

However, instead of supplying the target wallet's address, as done in `destination.address` for the transfer method, we need to provide an instance of [Address](../types/address.md) created from the deployed contract id.

If you have the [Contract](../contracts/) instance of the deployed contract, you can simply use its `id` property. However, if the contract was deployed with `forc deploy` or not by you, you will likely only have its ID in a hex string format. In such cases, you can create an [Address](../types/address.md) instance from the contract ID using `Address.fromAddressOrString('0x123...')`.

Here's an example demonstrating how to use `transferToContract`:

<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-4{ts:line-numbers}

Always remember to call the `waitForResult()` function on the transaction response. That ensures the transaction has been mined successfully before proceeding.

## Transferring Assets Between Wallets

> Transferring assets between wallets is straightforward within the SDK.

Reviewer: We need to keep only one section for transferring assets between accounts/wallets. It is either:

- [Transferring Assets Between Accounts](https://github.com/FuelLabs/fuels-ts/blob/d3637f7ec248fcdbdf37ca2b358909b34a001a28/apps/docs/src/guide/wallets/wallet-transferring.md#transferring-assets-between-accounts)

Or

- [Transferring Assets Between Wallets](https://github.com/FuelLabs/fuels-ts/blob/d3637f7ec248fcdbdf37ca2b358909b34a001a28/apps/docs/src/guide/wallets/wallet-transferring.md#transferring-assets-between-wallets)

Reviewer: Hey @mvares, let me know if this suggestion is not clear enough or if you have any further questions about it.

> # Wallet Transferring

Reviewer 2: I'm unsure how we should document `Account`-related methods. A predicate can transfer assets the same way as a wallet. Should we put it in some other section? Create a new `Accounts` section?
- Accounts
  - Transferring assets
  - Checking balances
  - etc.

> This guide provides instructions for transferring assets between wallets and contracts using the SDK. It includes methods for initiating transfers, handling transaction requests, and ensuring sufficient balance before transfers.

Reviewer 3: ```suggestion
This guide provides instructions for transferring assets between wallets and contracts using the SDK. It includes methods to validate balances and initiate and configure transfer requests.
```

## Transferring Assets Between Accounts

> The `account.transfer` method initiates a transaction request that transfers an asset from one wallet to another. This method requires three parameters:

Reviewer 2: ```suggestion
The `account.transfer` method creates and sends a transaction that transfers an asset from one account to another. This method requires three parameters:
```

> The `account.transfer` method initiates a transaction request that transfers an asset from one wallet to another. This method requires three parameters:

Reviewer 3: ```suggestion
The `transfer` method initiates a transaction request that transfers an asset from one wallet to another. This method requires three parameters:
```

I'd prefer to make this class agnostic due to the amount of inheritance we have and I don't think consumers will understand this complexity.

> 1. The receiver's wallet address.
> 2. The amount of the asset to be transferred.
> 3. The ID of the asset to be transferred (Optional - Defaults to the base asset id).

Reviewer 3: ```suggestion
1. The receiver's wallet address
2. The amount of the asset to be transferred
3. The ID of the asset to be transferred (optional - defaults to the base asset ID)
```

Nits

> Upon execution, this function returns a promise that resolves to a transaction response. To wait for the transaction to be processed, call `response.wait()`.

Reviewer 2: Will we stay with `.wait()` or `.waitForResult()`? #1146

Reviewer: @petertonysmith94 I reckon we should use `waitForResult` as it is the most used within our code examples.

### Example

> Here is an illustration of how to use the `account.transfer` function:

Reviewer 3: ```suggestion
Here is an example of how to use the `transfer` function:
```

<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-1{ts:line-numbers}

> In the previous example, we used the `transfer` method, which essentially creates a `ScriptTransactionRequest`, populates its data with the provided transfer information, and submits the transaction request to the node.

Reviewer 3: ```suggestion
In the previous example, we used the `transfer` method which creates a `ScriptTransactionRequest`, populates its data with the provided transfer information and submits the transaction.
```

However, there may be times when you need the Transaction ID before actually submitting it to the node. To achieve this, you can simply call the `createTransfer` method instead.

> This method also creates a `ScriptTransactionRequest` and populates it with the provided data, but instead of submitting it to the node, it returns the request.

Reviewer 3: ```suggestion
This method also creates a `ScriptTransactionRequest` and populates it with the provided data but returns the request object prior to submission.
```

<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-2{ts:line-numbers}

> Once you initiate a transaction request, any changes will create a new, separate transaction with a `different` ID. Therefore, avoid modifying the transaction request if you want to keep the same ID.

Reviewer 3: ```suggestion
> **Note**: Any changes made to a transaction request will alter the transaction ID. Therefore, you should only get the transaction ID after all modifications have been made.
```

Copied from the tx request docs for uniformity.

<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-3{ts:line-numbers}

## Transferring Assets To Contracts

> When transferring assets to a deployed contract, we use the `transferToContract` method, which shares a similar parameter structure with the `account.transfer` method.

Reviewer 3: ```suggestion
When transferring assets to a deployed contract, we use the `transferToContract` method, which shares a similar parameter structure with the `transfer` method.
```

> However, instead of supplying the target wallet's address, as done in `destination.address` for the transfer method, we need to provide an instance of [Address](../types/address.md) created from the deployed contract id.

Reviewer 3: This feels like poor DX? We should have consistency whether we're passing the Object or the literal str address 🤔

Reviewer 3: cc @FuelLabs/sdk-ts for thoughts?

Reviewer: @danielbate I remember we had discussions about this in the past. We then created an issue for it: https://github.com/FuelLabs/fuels-ts/issues/1584

Reviewer 4: There's also this:
 - https://github.com/FuelLabs/fuels-ts/issues/2348

If you have the [Contract](../contracts/) instance of the deployed contract, you can simply use its `id` property. However, if the contract was deployed with `forc deploy` or not by you, you will likely only have its ID in a hex string format. In such cases, you can create an [Address](../types/address.md) instance from the contract ID using `Address.fromAddressOrString('0x123...')`.

> ### Example

Reviewer 3: ```suggestion
```

Redundant title IMO
